core/src/main/java/feast/core/model/JobStatus.java

```diff
@@ -53,33 +51,41 @@
   /** job has been suspended */
   SUSPENDED;
 
-  private static final Collection<JobStatus> TERMINAL_STATE =
-      Collections.unmodifiableList(Arrays.asList(COMPLETED, ABORTED, ERROR));
+  private static final Set<JobStatus> TERMINAL_STATES = Set.of(COMPLETED, ABORTED, ERROR);
 
   /**
-   * Get a collection of terminal job state.
+   * Get the set of terminal job states.
    *
-   * <p>Terminal job state is final and will not change to any other state.
+   * <p>A terminal job state is final and will not change to any other state.
    *
-   * @return collection of terminal job state.
+   * @return set of terminal job states.
    */
-  public static Collection<JobStatus> getTerminalState() {
-    return TERMINAL_STATE;
+  public static Set<JobStatus> getTerminalStates() {
+    return TERMINAL_STATES;
```

Comment:
Got a little carried away, `JobStatus` changes are further than I meant to go for now… but I started touching `JobCoordinatorService` for `filter(job -> !job.hasTerminated())` and I ended up here…

These semantically are/should be Sets I think. I'm not sure I see value in these getters wrapping constants, after this PR their only remaining use is in a few tests. Considered removing them, but thought I'd leave it to discuss first.

Reply:
Makes sense.